### PR TITLE
create models and config database

### PR DESCRIPTION
**Pull request by VoTuan232**
Create models and config database

### CODE REVIEW

**Comment by VoTuan232** [comment]
ready

File: app/Models/Category.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Category extends Model
+{
+	protected $fillable = [
```
Comment (BunnyPi04) [inline app/Models/Category.php]: tab lại cả file
https://viblo.asia/p/cac-convention-thuong-gap-khi-moi-bat-dau-training-php-tai-framgia-bJzKmgq6l9N#_21-tab---indent-6

File: app/Models/Role.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    public function users()
+    {
+    	return $this->belongsToMany('App\Models\User', 'role_user');
+    }
+
+    public function hasAccess(array $permissions):bool
+    {
+        foreach($permissions as $permission)
+        {
+        	if($this->hasPermission($permission))
```
Comment (BunnyPi04) [inline app/Models/Role.php]: `Cần phải có một dấu cách phía sau những từ khoá Control structure (như if, else, for ...)`

https://viblo.asia/p/cac-convention-thuong-gap-khi-moi-bat-dau-training-php-tai-framgia-bJzKmgq6l9N#_24-khoang-trang-whitespace-9

File: app/Models/User.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        'name', 'email', 'password',
+    ];
+
+    /**
+     * The attributes that should be hidden for arrays.
+     *
+     * @var array
+     */
+    protected $hidden = [
+        'password', 'remember_token',
```
Comment (BunnyPi04) [inline app/Models/User.php]: chú ý convention về mảng
https://viblo.asia/p/cac-convention-thuong-gap-khi-moi-bat-dau-training-php-tai-framgia-bJzKmgq6l9N#_27-mang-va-chuoi-12

File: app/Models/User.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public function inRole(string $roleSlug)
+    {
+        return $this->roles()->where('slug', $roleSlug)->count()==1;
```
Comment (BunnyPi04) [inline app/Models/User.php]: convention
https://viblo.asia/p/cac-convention-thuong-gap-khi-moi-bat-dau-training-php-tai-framgia-bJzKmgq6l9N#_24-khoang-trang-whitespace-9

**Comment by BunnyPi04** [comment]
commented

File: app/Models/User.php [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public function inRole(string $roleSlug)
+    {
+        return $this->roles()->where("slug", $roleSlug)->count()==1;
```
Comment (BunnyPi04) [inline app/Models/User.php]: convention
https://viblo.asia/p/cac-convention-thuong-gap-khi-moi-bat-dau-training-php-tai-framgia-bJzKmgq6l9N#_24-khoang-trang-whitespace-9
https://viblo.asia/p/cac-convention-thuong-gap-khi-moi-bat-dau-training-php-tai-framgia-bJzKmgq6l9N#_27-mang-va-chuoi-12

**Comment by BunnyPi04** [comment]
commented

**Comment by VoTuan232** [comment]
ready

File: app/Models/User.php [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public function inRole(string $roleSlug)
+    {
+        return $this->roles()->where("slug", $roleSlug)->count() == 1;
```
Comment (BunnyPi04) [inline app/Models/User.php]: sử dụng nháy đơn nhé

**Comment by BunnyPi04** [comment]
commented

File: app/Models/User.php (hunk 6 above)
Comment (VoTuan232) [reply app/Models/User.php]: vâng ạ :3

**Comment by VoTuan232** [comment]
ready

**Pull request merged**